cloudimg/ms_azure.py

```diff
@@ -0,0 +1,480 @@
+from datetime import datetime, timedelta
+import logging
+import time
+
+import attr
+
+from azure.storage.blob import (
+    AccountSasPermissions,
+    BlobServiceClient,
+    BlobSasPermissions,
+    generate_account_sas,
+    generate_blob_sas,
+    ResourceTypes,
+)
+
+from cloudimg.common import BaseService, PublishingMetadata
+
+log = logging.getLogger(__name__)
+
+
+def get_values_from_connection_string(connection_str):
```

Comment:
Could you please clarify which of the functions and classes added here are intended to be public API?  i.e. you expect that someone outside the repo should import and call them?

If there are things in here that are intended as internal helpers only and not public API, please make it obvious in their names, e.g. name them with a leading underscore.  Or move them to a separate module whose name starts with a leading underscore.

When maintaining libraries I take backwards-compatibility seriously, so if you add these without marking them as private somehow and then want to make backwards-incompatible changes to them in later merge requests, I'll be complaining... it's better to add only the bare minimum as public API.

Reply:
I've changed these helpers to have the leading underscore before their name thus indicating they're not supposed to be public API, thanks a lot for your comment!